doc/advanced-configuration/outgoing-connections.md

```diff
@@ -188,278 +103,143 @@ sslmode     = verify-full
-
-Using SSL for Riak connection requires passing extra options in `ConnectionOptions` to the
-aforementioned `riak` tuple.
+#### `outgoing_pools.riak.*.connection.address`
+* **Syntax:** string
+* **Example:** `address = "127.0.0.1"`
 
-Here is the proper syntax:
+#### `outgoing_pools.riak.*.connection.port`
+* **Syntax:** integer
+* **Example:** `port = 8087`
 
-* **Credentials**
-    * **Description:** Specifies credentials to use to connect to the database.
-    * **Syntax:** `{credentials, User, Password}`
-    * **Supported values** `User` and `Password` are strings with a database username and password respectively.
+#### `outgoing_pools.riak.*.connection.credentials`
+* **Syntax:** `{user = "username", password = "pass}`
+* **Default:** none
+* **Example:** `credentials = {user = "myuser", password = "tisismepasswd"}`
+* **Comment:** optional - setting this option forces connection over TLS
 
-* **CACert**
-    * **Description:** Specifies a path to the CA certificate that was used to sign the database certificates.
-    * **Syntax:** `{cacertfile, Path}`
-    * **Supported values** `Path` is a string with a path to the CA certificate file.
+Riak supports also all TLS-specific options described in TLS section.
 
-* **SSL_Opts**
-    * **Description**: list of SSL options as defined in [Erlang SSL module doc](http://erlang.org/doc/man/ssl.html)
-      They will be passed verbatim to the `ssl:connect` function.
-    * **Syntax** `{ssl_opts, ListOfSSLOpts}`
-    * **Example**:
+## Cassandra options
 
-```erlang
-{ssl_opts, [{ciphers, ["AES256-SHA", "DHE-RSA-AES128-SHA256"]},
-            {server_name_indication, disable}]}
-```
+#### `outgoing_pools.cassandra.*.connection.servers`
+* **Syntax:** a list of maps containing keys `"ip_adddress"` and `"port"`
+* **Default:** `[{ip_address = "localhost", port = 9042}]`
+* **Example:** `servers = [{ip_address = "host_one", port = 9042}, {ip_address = "host_two", port = 9042}]`
 
-##### Example configuration
+#### `outgoing_pools.cassandra.*.connection.keyspace`
+* **Syntax:** string
+* **Default:** `"mongooseim"`
+* **Example:** `keyspace = "big_mongooseim_database"`
 
-An example configuration can look as follows:
+#### `outgoing_pools.cassandra.*.connection.auth`
+* **Syntax:** {module = string, options = list of pairs of strings}
+* **Default:** undefined (no auth)
+* **Example:** `auth = {module = "cqerl_auth_plain_handler", options = [["test", "aaa"]]}`
 
-```erlang
-{outgoing_pools, [
- {riak, global, default, [{workers, 20}, {strategy, next_worker}],
-   [{address, "127.0.0.1"}, {port, 8087},
-    {credentials, "username", "pass"},
-    {cacertfile, "path/to/cacert.pem"}]}
-]}.
-```
-
-## Cassandra connection setup
-
-The minimum pool definition for cassandra workers looks as follows:
-
-```erlang
-{outgoing_pools, [
- {cassandra, global, default, [], []}
-]}.
-```
+Internally, the list of pairs is transformed into a proplist, so the above would reach Cassandra client driver as:
```

Comment:
Do we want to have a sentence like this in the docs at all? Maybe it would be better just to say:
```suggestion
The above would reach Cassandra client driver as:
```